modules/travel_pay/spec/services/appointments_client_spec.rb

```diff
@@ -0,0 +1,179 @@
+        .and_return(*tokens)
+    end
+
+    it 'returns a response only with appointments with no claims' do
+      @stubs.get('/api/v1.1/appointments?excludeWithClaims=true') do
+        [
+          200,
+          {},
+          {
+            'data' => [
+              {
+                'id' => 'uuid1',
+                'appointmentSource' => 'API',
+                'appointmentDateTime' => '2024-01-01T16:45:34.465Z',
+                'appointmentName' => 'string',
+                'appointmentType' => 'EnvironmentalHealth',
+                'facilityName' => 'Cheyenne VA Medical Center',
+                'serviceConnectedDisability' => 30,
+                'currentStatus' => 'string',
+                'appointmentStatus' => 'Completed',
+                'externalAppointmentId' => '12345678-0000-0000-0000-000000000001',
+                'associatedClaimId' => nil,
+                'associatedClaimNumber' => nil,
+                'isCompleted' => true
+              },
+              {
+                'id' => 'uuid2',
+                'appointmentSource' => 'API',
+                'appointmentDateTime' => '2024-03-01T16:45:34.465Z',
+                'appointmentName' => 'string',
+                'appointmentType' => 'EnvironmentalHealth',
+                'facilityName' => 'Cheyenne VA Medical Center',
+                'serviceConnectedDisability' => 30,
+                'currentStatus' => 'string',
+                'appointmentStatus' => 'Completed',
+                'externalAppointmentId' => '12345678-0000-0000-0000-000000000002',
+                'associatedClaimId' => nil,
+                'associatedClaimNumber' => nil,
+                'isCompleted' => true
+              },
+              {
+                'id' => 'uuid3',
+                'appointmentSource' => 'API',
+                'appointmentDateTime' => '2024-02-01T16:45:34.465Z',
+                'appointmentName' => 'string',
+                'appointmentType' => 'EnvironmentalHealth',
+                'facilityName' => 'Cheyenne VA Medical Center',
+                'serviceConnectedDisability' => 30,
+                'currentStatus' => 'string',
+                'appointmentStatus' => 'Completed',
+                'externalAppointmentId' => '12345678-0000-0000-0000-000000000003',
+                'associatedClaimId' => nil,
+                'associatedClaimNumber' => nil,
+                'isCompleted' => true
+              }
+            ]
+          }
+        ]
+      end
+
+      expected_ids = %w[uuid1 uuid2 uuid3]
+
+      client = TravelPay::AppointmentsClient.new
+      appts_response = client.get_all_appointments(*tokens, { 'excludeWithClaims' => true })
+      actual_appt_ids = appts_response.body['data'].pluck('id')
+
+      expect(actual_appt_ids).to eq(expected_ids)
+    end
+
+    it 'returns a response with all appointments' do
+      @stubs.get('/api/v1.1/appointments') do
+        [
+          200,
+          {},
+          {
+            'data' => [
```

Comment:
Unless there's a difference in the data that I'm missing, let's see if we can move this to the context scope.